File: modules/wem-api/src/main/java/com/enonic/wem/api/security/acl/Permission.java
package com.enonic.wem.api.security.acl;

public enum Permission
{

    READ,
    CREATE,
    MODIFY,
    DELETE,
    PUBLISH,
    READ_PERMISSIONS,
    WRITE_PERMISSIONS;

    private final String id;

    Permission()
    {
        this.id = this.toString().toLowerCase();
    }

    public String id()
    {
        return id;
    }

}

File: modules/admin/admin-ui/src/main/resources/web/admin/common/js/security/acl/Permission.ts
module api.security.acl {

    /**
     *  enum Color{
     *      Red, Green
     *  }
     *
     *  // To String
     *  var green: string = Color[Color.Green];
     *
     *  // To Enum / number
     *  var color : Color = Color[green];
     */

    export enum Permission {
        READ,
        CREATE,
        MODIFY,
        DELETE,
        PUBLISH,
        READ_PERMISSIONS,
        WRITE_PERMISSIONS
    }

    export enum PermissionState {
        ALLOW,
        DENY,
        INHERIT
    }
}
